tools/Azure.SDK.ChangelogGen/Azure.SDK.ChangelogGen/ChangeLogResult.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.SDK.ChangelogGen.Compare;
+using Azure.SDK.ChangelogGen.Report;
+using Azure.SDK.ChangelogGen.Utilities;
+
+namespace Azure.SDK.ChangelogGen
+{
+    public class ChangeLogResult
+    {
+        public ChangeSet? ApiChange { get; set; } = null;
+        public StringValueChange? SpecVersionChange { get; set; } = null;
+        public StringValueChange? AzureCoreVersionChange { get; set; } = null;
+        public StringValueChange? AzureResourceManagerVersionChange { get; set; } = null;
+
+        public Release GenerateReleaseNote(string version, string date, List<ChangeCatogory> filter)
+        {
+            string PREFIX = "- ";
+            Release report = new Release(version, date);
+
+            if (ApiChange?.GetBreakingChanges().Count() > 0)
+            {
+                ReleaseNoteGroup breakingGroup = new ReleaseNoteGroup("Breaking Changes");
```

Comment:
Not quite sure I understand the comment correct here. Do you mean we will have a doc to demonstrate how to mitigate breaking change? 

This part code is to handle the case when breaking change detected just in case though we dont expect any breaking change. In case there are breaking changes, the tool will log an error as well as adding these breaking change into changelog under "Breaking Changes" section anyway to make sure people are aware of these unexpected breaking changes and fix them when reviewing PR.

Reply:
> Do you mean we will have a doc to demonstrate how to mitigate breaking change?

Yes.  That is the standard for Azure SDK packages.  [[Example](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/BREAKING_CHANGES.md)].   Here, again, I don't know if there was an exception granted to management libraries.   @ArthurMa1978?

Reply:
Once the management plane SDK reaches GA, no new releases will introduce breaking changes. Any breaking changes in the swagger spec will be handled by using customization code (or autogen code in the future).
So just skip this part here.

Reply:
@jsquire , kindly ping..., please let us know if you have further questions about this. thanks.

Reply:
> Once the management plane SDK reaches GA, no new releases will introduce breaking changes. Any breaking changes in the swagger spec will be handled by using customization code (or autogen code in the future).
So just skip this part here.

We generally do not document API breaking changes from beta-to-beta.   I thought we agreed to only list these for GA libraries - so if this section does not apply to GA libraries, then let's not include it or, let's at least have some minimal plan for how we'll communicate to partner teams that they need to put together a breaking changes guide.

Reply:
I misunderstood that we will fix breaking change for beta too which seems not true. fixed to ignore breaking changes for beta version. Thanks.

Reply:
Can you help me understand where that logic is?  I don't see a filter for breaking changes or obsolete against beta versions.

Reply:
the original logic was when calling GenerateReleaseNote method in ChangeLogResult.cs, it took parameters 'ignoreBreakingChange' and 'filter'. the 'ignoreBreakingChange' would be true for preview release to skip them while only ObsoleteChange would be set in filter to only log obsoleted changes. Just now, I realize that the api comparison part is just not needed at all for preview version, so i have updated the code to skip that part for preview version to make the logic more clean and bring better perf and stability. thx.